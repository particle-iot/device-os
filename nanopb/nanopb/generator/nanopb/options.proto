// This is a transitional file, to provide parallel support between the old
// nanopb.proto and new options.proto files. Eventually nanopb.proto will
// be left only for legacy code, but for now the generator is still also
// using it. However, your new code can start using this file already now.
// See pull request #241 for details:
// https://github.com/nanopb/nanopb/pull/241

// Custom options for defining:
// - Maximum size of string/bytes
// - Maximum number of elements in array
//
// These are used by nanopb to generate statically allocable structures
// for memory-limited environments.

syntax = "proto2";
import "google/protobuf/descriptor.proto";

package nanopb;
option java_package = "fi.kapsi.koti.jpa.nanopb";

enum FieldType {
    FT_DEFAULT = 0; // Automatically decide field type, generate static field if possible.
    FT_CALLBACK = 1; // Always generate a callback field.
    FT_POINTER = 4; // Always generate a dynamically allocated field.
    FT_STATIC = 2; // Generate a static field or raise an exception if not possible.
    FT_IGNORE = 3; // Ignore the field completely.
    FT_INLINE = 5; // Legacy option, use the separate 'fixed_length' option instead
}

enum IntSize {
    IS_DEFAULT = 0; // Default, 32/64bit based on type in .proto
    IS_8 = 8;
    IS_16 = 16;
    IS_32 = 32;
    IS_64 = 64;
}

// This is the inner options message, which basically defines options for
// a field. When it is used in message or file scope, it applies to all
// fields.
message Options {
  // Allocated size for 'bytes' and 'string' fields.
  // For string fields, this should include the space for null terminator.
  optional int32 max_size = 1;

  // Maximum length for 'string' fields. Setting this is equivalent
  // to setting max_size to a value of length+1.
  optional int32 max_length = 14;

  // Allocated number of entries in arrays ('repeated' fields)
  optional int32 max_count = 2;

  // Size of integer fields. Can save some memory if you don't need
  // full 32 bits for the value.
  optional IntSize int_size = 7 [default = IS_DEFAULT];

  // Force type of field (callback or static allocation)
  optional FieldType type = 3 [default = FT_DEFAULT];

  // Use long names for enums, i.e. EnumName_EnumValue.
  optional bool long_names = 4 [default = true];

  // Add 'packed' attribute to generated structs.
  // Note: this cannot be used on CPUs that break on unaligned
  // accesses to variables.
  optional bool packed_struct = 5 [default = false];

  // Add 'packed' attribute to generated enums.
  optional bool packed_enum = 10 [default = false];

  // Skip this message
  optional bool skip_message = 6 [default = false];

  // Generate oneof fields as normal optional fields instead of union.
  optional bool no_unions = 8 [default = false];

  // integer type tag for a message
  optional uint32 msgid = 9;

  // decode oneof as anonymous union
  optional bool anonymous_oneof = 11 [default = false];

  // Proto3 singular field does not generate a "has_" flag
  optional bool proto3 = 12 [default = false];

  // Generate an enum->string mapping function (can take up lots of space).
  optional bool enum_to_string = 13 [default = false];

  // Generate bytes arrays with fixed length
  optional bool fixed_length = 15 [default = false];
}

// Extensions to protoc 'Descriptor' type in order to define options
// inside a .proto file.
//
// Protocol Buffers extension number registry
// --------------------------------
// Project:  Nanopb
// Contact:  Petteri Aimonen <jpa@kapsi.fi>
// Web site: http://kapsi.fi/~jpa/nanopb
// Extensions: 1010 (all types)
// --------------------------------

extend google.protobuf.FileOptions {
    optional Options fileopt = 1010;
}

extend google.protobuf.MessageOptions {
    optional Options msgopt = 1010;
}

extend google.protobuf.EnumOptions {
    optional Options enumopt = 1010;
}

extend google.protobuf.FieldOptions {
    optional Options fieldopt = 1010;
}


