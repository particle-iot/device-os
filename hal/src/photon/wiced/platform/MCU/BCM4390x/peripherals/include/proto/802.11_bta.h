/*
 * Broadcom Proprietary and Confidential. Copyright 2016 Broadcom
 * All Rights Reserved.
 *
 * This is UNPUBLISHED PROPRIETARY SOURCE CODE of Broadcom Corporation;
 * the contents of this file may not be disclosed to third parties, copied
 * or duplicated in any form, in whole or in part, without the prior
 * written permission of Broadcom Corporation.
 *
 * BT-AMP (BlueTooth Alternate Mac and Phy) 802.11 PAL (Protocol Adaptation Layer)
 *
 * $Id: 802.11_bta.h 382882 2013-02-04 23:24:31Z xwei $
*/

#ifndef _802_11_BTA_H_
#define _802_11_BTA_H_

#define BT_SIG_SNAP_MPROT		"\xAA\xAA\x03\x00\x19\x58"

/* BT-AMP 802.11 PAL Protocols */
#define BTA_PROT_L2CAP				1
#define	BTA_PROT_ACTIVITY_REPORT		2
#define BTA_PROT_SECURITY			3
#define BTA_PROT_LINK_SUPERVISION_REQUEST	4
#define BTA_PROT_LINK_SUPERVISION_REPLY		5

/* BT-AMP 802.11 PAL AMP_ASSOC Type IDs */
#define BTA_TYPE_ID_MAC_ADDRESS			1
#define BTA_TYPE_ID_PREFERRED_CHANNELS		2
#define BTA_TYPE_ID_CONNECTED_CHANNELS		3
#define BTA_TYPE_ID_CAPABILITIES		4
#define BTA_TYPE_ID_VERSION			5
#endif /* _802_11_bta_h_ */
