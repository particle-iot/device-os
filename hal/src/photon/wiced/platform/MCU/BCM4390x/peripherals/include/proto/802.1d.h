/*
 * Broadcom Proprietary and Confidential. Copyright 2016 Broadcom
 * All Rights Reserved.
 *
 * This is UNPUBLISHED PROPRIETARY SOURCE CODE of Broadcom Corporation;
 * the contents of this file may not be disclosed to third parties, copied
 * or duplicated in any form, in whole or in part, without the prior
 * written permission of Broadcom Corporation.
 *
 * Fundamental types and constants relating to 802.1D
 *
 * $Id: 802.1d.h 382882 2013-02-04 23:24:31Z xwei $
 */

#ifndef _802_1_D_
#define _802_1_D_

/* 802.1D priority defines */
#define	PRIO_8021D_NONE		2	/* None = - */
#define	PRIO_8021D_BK		1	/* BK - Background */
#define	PRIO_8021D_BE		0	/* BE - Best-effort */
#define	PRIO_8021D_EE		3	/* EE - Excellent-effort */
#define	PRIO_8021D_CL		4	/* CL - Controlled Load */
#define	PRIO_8021D_VI		5	/* Vi - Video */
#define	PRIO_8021D_VO		6	/* Vo - Voice */
#define	PRIO_8021D_NC		7	/* NC - Network Control */
#define	MAXPRIO			7	/* 0-7 */
#define NUMPRIO			(MAXPRIO + 1)

#define ALLPRIO		-1	/* All prioirty */

/* Converts prio to precedence since the numerical value of
 * PRIO_8021D_BE and PRIO_8021D_NONE are swapped.
 */
#define PRIO2PREC(prio) \
	(((prio) == PRIO_8021D_NONE || (prio) == PRIO_8021D_BE) ? ((prio^2)) : (prio))

#endif /* _802_1_D__ */
