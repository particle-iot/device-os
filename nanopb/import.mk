NANOPB_MODULE_NAME = nanopb
NANOPB_MODULE_PATH ?= $(PROJECT_ROOT)/$(NANOPB_MODULE_NAME)
include $(call rwildcard,$(NANOPB_MODULE_PATH)/,include.mk)

NANOPB_BUILD_PATH_EXT = $(BUILD_TARGET_PLATFORM)
NANOPB_LIB_DIR = $(BUILD_PATH_BASE)/$(NANOPB_MODULE_NAME)/$(NANOPB_BUILD_PATH_EXT)
NANOPB_LIB_DEP = $(NANOPB_LIB_DIR)/lib$(NANOPB_MODULE_NAME).a
