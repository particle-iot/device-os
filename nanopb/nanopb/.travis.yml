# Travis CI has no ability to handle 3 langauges (c, c++, python)
# and it overrides $CC/$CXX if language is set to c/c++ (only one, not both).
#
# Set language to python since at least the result of that is something useful.
language: python

python:
    - "2.7"
    - "3.4"

# Manage the C/C++ compiler manually
env:
    - CC=gcc     CXX=g++
    - CC=gcc-4.8 CXX=g++-4.8
    - CC=gcc-4.9 CXX=g++-4.9
    - CC=gcc-5   CXX=g++-5
    - CC=clang   CXX=clang++

addons:
    apt:
        sources:
            - ubuntu-toolchain-r-test
        packages:
            - gcc-4.8
            - g++-4.8
            - gcc-4.9
            - g++-4.9
            - gcc-5
            - g++-5


before_install:
    - export PATH=$HOME/.local/bin:$HOME/protobuf/bin:$PATH
    - export MAKEFLAGS=-j$(nproc)
    - $CC --version
    - $CXX --version
    - python --version
    - lsb_release -a

# Seems to be issues with concurrent builds
#cache:
#    directories:
#        - $HOME/protobuf

# Rather then compile protobuf 3 from source, use the binaries now available
# to speed up build time and reduce surprises until Ubuntu adds protobuf3
# packages to the repository.
install:
    - mkdir -p $HOME/protobuf && pushd $HOME/protobuf
      && curl -LO 'https://github.com/google/protobuf/releases/download/v3.4.0/protoc-3.4.0-linux-x86_64.zip'
      && unzip protoc-3.4.0-linux-x86_64.zip
      && popd
    - curl -L 'https://github.com/google/protobuf/releases/download/v3.4.0/protobuf-python-3.4.0.tar.gz' | tar xzf -
      && pushd protobuf-3.4.0/python
      && python setup.py build && python setup.py install
      && popd

script:
    - pushd generator/proto && make && popd
    - pushd tests && scons CC=$CC CXX=$CXX && popd
