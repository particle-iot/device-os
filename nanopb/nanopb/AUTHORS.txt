Petteri Aimonen <jpa@npb.mail.kapsi.fi>
Michael Poole <mdpoole@troilus.org>
Daniel Kan <extremeblue99@gmail.com>
Stan Hu <stanhu@aclimalabs.com>
David Hotham <david.hotham@blueyonder.co.uk>
Steffen Siering <steffen siering gmail com>
Jens Steinhauser <jens.steinhauser@gmail.com>
Pavel Ilin <ilin.pa@gmail.com>
Kent Ryhorchuk <kryhorchuk@xeralux.com>
Martin Donath <scifish@gmail.com>
Oliver Lee <oliverzlee@gmail.com>
Michael Haberler <git@mah.priv.at>
Nicolas Colomer <ncolomer@viadeoteam.com>
Ivan Kravets <me@ikravets.com>
Kyle Manna <kyle@kylemanna.com>
Benjamin Kamath <ben.kamath@synapse.com>
Andrew Ruder <andrew.ruder@elecsyscorp.com>
Kenshi Kawaguchi <kenshi@recurse.ca>
isotes <isotes@gmail.com>
Maxim Khitrov <max@mxcrypt.com>
Yaniv Mordekhay <yanivmo@users.noreply.github.com>
Ming Zhao <mzhao@luminatewireless.com>
Google, Inc.
Tom Roeder <tmroeder@google.com>
Piotr Sikora <piotrsikora@google.com>
Bernhard KraÌˆmer <bdkrae@gmail.com>
Konstantin Podsvirov <konstantin@podsvirov.pro>
William A. Kennington III <wak@google.com>
Guillaume Lager <g.lager@innoseis.com>
Tobias Haegermarck <tobias.haegermarck@gmail.com>
Justin DeMartino <jdemarti@gmail.com>
Constantine Grantcharov <cgrantcharov@trustpointinnovation.com>
Nick Ewalt <nicholasewalt@google.com>
Harald Fernengel <harryf@gmx.com>
Alice Wang <aw@squareup.com>
Kevin Fitch <kfitch42@gmail.com>
Kamal Marhubi <kamal@marhubi.com>
Elco Jacobs <elco@brewpi.com>

