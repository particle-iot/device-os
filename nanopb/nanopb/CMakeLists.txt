cmake_minimum_required(VERSION 2.8)

project(nanopb C)

set(nanopb_VERSION_STRING nanopb-0.3.9)

string(REPLACE "nanopb-" "" nanopb_VERSION ${nanopb_VERSION_STRING})

option(nanopb_BUILD_RUNTIME "Build the headers and libraries needed at runtime" ON)
option(nanopb_BUILD_GENERATOR "Build the protoc plugin for code generation" ON)
option(nanopb_MSVC_STATIC_RUNTIME "Link static runtime libraries" ON)

if(NOT DEFINED CMAKE_DEBUG_POSTFIX)
    set(CMAKE_DEBUG_POSTFIX "d")
endif()

include(GNUInstallDirs)

if(MSVC AND nanopb_MSVC_STATIC_RUNTIME)
    foreach(flag_var
            CMAKE_C_FLAGS CMAKE_C_FLAGS_DEBUG CMAKE_C_FLAGS_RELEASE
            CMAKE_C_FLAGS_MINSIZEREL CMAKE_C_FLAGS_RELWITHDEBINFO)
        if(${flag_var} MATCHES "/MD")
            string(REGEX REPLACE "/MD" "/MT" ${flag_var} "${${flag_var}}")
        endif(${flag_var} MATCHES "/MD")
    endforeach(flag_var)
endif()

if(NOT DEFINED CMAKE_INSTALL_CMAKEDIR)
    set(CMAKE_INSTALL_CMAKEDIR "lib/cmake/nanopb")
endif()

if(nanopb_BUILD_GENERATOR)
    set(generator_protos nanopb plugin)

    find_package(PythonInterp 2.7 REQUIRED)
    execute_process(
        COMMAND ${PYTHON_EXECUTABLE} -c
            "from distutils import sysconfig; print(sysconfig.get_python_lib(prefix='${CMAKE_INSTALL_PREFIX}'))"
        OUTPUT_VARIABLE PYTHON_INSTDIR
        OUTPUT_STRIP_TRAILING_WHITESPACE
    )

    foreach(generator_proto IN LISTS generator_protos)
        string(REGEX REPLACE "([^;]+)" "${PROJECT_SOURCE_DIR}/generator/proto/\\1.proto" generator_proto_file "${generator_proto}")
        string(REGEX REPLACE "([^;]+)" "\\1_pb2.py" generator_proto_py_file "${generator_proto}")
        add_custom_command(
            OUTPUT ${generator_proto_py_file}
            COMMAND protoc --python_out=${PROJECT_BINARY_DIR} -I${PROJECT_SOURCE_DIR}/generator/proto ${generator_proto_file}
            DEPENDS ${generator_proto_file}
        )
        add_custom_target("generate_${generator_proto_py_file}" ALL DEPENDS ${generator_proto_py_file})
        install(
            FILES ${PROJECT_BINARY_DIR}/${generator_proto_py_file}
			DESTINATION ${PYTHON_INSTDIR}
        )
    endforeach()
endif()

if(nanopb_BUILD_RUNTIME)
    add_library(protobuf-nanopb STATIC
        pb.h
        pb_common.h
        pb_common.c
        pb_encode.h
        pb_encode.c
        pb_decode.h
        pb_decode.c)

    target_include_directories(protobuf-nanopb INTERFACE
      $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
    )

    configure_file(extra/nanopb-config-version.cmake.in
        nanopb-config-version.cmake @ONLY)

    install(TARGETS protobuf-nanopb EXPORT nanopb-targets
        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})

    install(EXPORT nanopb-targets
        DESTINATION ${CMAKE_INSTALL_CMAKEDIR}
        NAMESPACE nanopb::)

    install(FILES extra/nanopb-config.cmake
        ${CMAKE_CURRENT_BINARY_DIR}/nanopb-config-version.cmake
        DESTINATION ${CMAKE_INSTALL_CMAKEDIR})

    install(FILES pb.h pb_common.h pb_encode.h pb_decode.h
        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
endif()
