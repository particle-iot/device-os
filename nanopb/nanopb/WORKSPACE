workspace(name="com_github_nanopb_nanopb")
