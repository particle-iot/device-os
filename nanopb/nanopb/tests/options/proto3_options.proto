syntax = "proto3";

import "nanopb.proto";

message Message1
{
    int32 proto3_default = 1;
    int32 proto3_off = 2 [(nanopb).proto3 = false];
    int32 proto3_on = 3 [(nanopb).proto3 = true];
}

