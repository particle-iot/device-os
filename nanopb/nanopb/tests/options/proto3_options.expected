! bool has_proto3_default
bool has_proto3_off
! bool has_proto3_on

